Lib/test/test_zipfile.py

```diff
@@ -1727,6 +1727,17 @@ def test_empty_file_raises_BadZipFile(self):
             fp.write("short file")
         self.assertRaises(zipfile.BadZipFile, zipfile.ZipFile, TESTFN)
 
+    def test_negative_central_directory_offset_raises_BadZipFile(self):
+        # Zip file containing an empty EOCD record
+        buffer = [80, 75, 5, 6, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+
+        # Set the size of the central directory bytes to become 1,
+        # causing the central directory offset to become negative
+        buffer[12] = 1
+
+        f = io.BytesIO(bytes(buffer))
```

Comment:
... and no need to convert it to bytes here.
```suggestion
        f = io.BytesIO(bytes(buffer))
        f = io.BytesIO(buffer)
```